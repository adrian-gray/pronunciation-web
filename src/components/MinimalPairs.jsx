import React from 'react'

import {
  Paper,
  Table,
  TableBody,
  TableCell,
  TableRow,
  Typography,
  withStyles
} from '@material-ui/core'

import SplitHilite from './SplitHilite'

const styles = (theme) => ({
  cell: {
    paddingLeft: 12,
    paddingRight: 12,
    textAlign: 'center'
  },
  largeText: theme.largeText,
  headspace: theme.headspace
})

const MinimalPairs = (props) => {
  const { classes, ipa, pairs, tag } = props

  const rows = pairs.map((pair, key) => (
    <TableRow key={key}>
      <TableCell className={classes.cell} key={0}>
        <Typography className={classes.largeText}>
          <SplitHilite str={pair[0]} />
        </Typography>
      </TableCell>
      <TableCell className={classes.cell} key={1}>
        <Typography className={classes.largeText}>
          <SplitHilite str={pair[1]} />
        </Typography>
      </TableCell>
    </TableRow>
  ))

  let display
  if (props.userAuth) {
    display = (
      <Paper>
        <Table>
          <TableBody>
            {rows}
          </TableBody>
        </Table>
      </Paper>
    )
  } else {
    display = (
      <Typography variant='subheading' className={classes.headspace} gutterBottom>
        {'Sorry, activity for members only.'}
      </Typography>
    )
  }

  return (
    <div className={classes.headspace} >
      <Typography variant='title' gutterBottom>
        {`English  minimal pairs with ${tag} - `}
        <SplitHilite str={ipa} />
      </Typography>
      {display}
    </div>
  )
}

export default withStyles(styles)(MinimalPairs)
