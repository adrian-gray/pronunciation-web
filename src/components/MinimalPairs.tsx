import React from "react";

import Container from "react-bootstrap/Container";
import Table from "react-bootstrap/Table";

import SplitHilite from "./SplitHilite";

export default (props: {
  ipa: string;
  pairs: string[][];
  tag: string;
  isUserAuth: boolean;
}) => {
  const { ipa, pairs, tag } = props;

  const rows = pairs.map((pair, key) => (
    <tr key={key}>
      <td className="cell" key={0}>
        <p className="largeText">
          <SplitHilite str={pair[0]} />
        </p>
      </td>
      <td className="cell" key={1}>
        <p className="largeText">
          <SplitHilite str={pair[1]} />
        </p>
      </td>
    </tr>
  ));

  return (
    <div className="headspace">
      <h3>
        {`English  minimal pairs with ${tag} - `}
        <SplitHilite str={ipa} />
      </h3>
      <Container>
        <Table>
          <tbody>{rows}</tbody>
        </Table>
      </Container>
    </div>
  );
};
