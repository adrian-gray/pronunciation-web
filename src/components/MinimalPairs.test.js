import React from "react";
import { render, cleanup } from "@testing-library/react";

import MinimalPairs from "./MinimalPairs";

afterEach(cleanup);

test("MinimalPairs", async () => {
  const props = {
    ipa: "a:",
    tag: "dork",
    pairs: [["fat", "cat"], ["happy", "dog"]],
    isUserAuth: true
  };

  const { container } = render(<MinimalPairs {...props} />);

  expect(container).toMatchSnapshot();
});
