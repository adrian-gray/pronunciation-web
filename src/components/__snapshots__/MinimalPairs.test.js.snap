// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinimalPairs 1`] = `
<div>
  <div
    class="headspace"
  >
    <h3>
      English  minimal pairs with dork - 
      <span>
        a:
      </span>
       
    </h3>
    <div>
      <table
        class="table"
      >
        <tbody>
          <tr>
            <td
              class="cell"
            >
              <p
                class="largeText"
              >
                <span>
                  fat
                </span>
                 
              </p>
            </td>
            <td
              class="cell"
            >
              <p
                class="largeText"
              >
                <span>
                  cat
                </span>
                 
              </p>
            </td>
          </tr>
          <tr>
            <td
              class="cell"
            >
              <p
                class="largeText"
              >
                <span>
                  happy
                </span>
                 
              </p>
            </td>
            <td
              class="cell"
            >
              <p
                class="largeText"
              >
                <span>
                  dog
                </span>
                 
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
